applications/RANSApplication/custom_utilities/rans_variable_difference_norm_calculation_utility.cpp

```diff
@@ -0,0 +1,97 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+#include <tuple>
+
+// External includes
+
+// Project includes
+#include "includes/communicator.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+
+// Application includes
+
+// Include base h
+#include "rans_variable_difference_norm_calculation_utility.h"
+
+namespace Kratos
+{
+template <typename TDataType>
+void RansVariableDifferenceNormsCalculationUtility<TDataType>::InitializeCalculation()
```

Comment:
I assume that `InitializeCalculation()` is needed in the 99% of the times so I'll encapsulate it within the `CalculateDifferenceNorm()`  such that the user doesn't need to do two calls to do one operation. We can add an extra `InitializeValues = False` argument.

Reply:
This is to be used to calculate convergence of a variable (outside of `ConvergenceCriteria`) in the same time step. Therefore, an object of this needs to call `InitializeCalculation` when non-linear iterations starts, and after each non-linear iteration `CalculateDifferenceNorm` can be called so that convergence of that variable can be evaluated. It is required to evaluate coupling iterations for coupled `Strategy` in `Formulations`. Hence I am not sure how to integrate this in to one method call.

Reply:
OK. To avoid changing the way you check convergence let's keep it as it is. In the future we can think about a `VariableNormCalculationUtility`.